lms/djangoapps/courseware/tests/test_date_summary.py

```diff
@@ -586,9 +586,10 @@ def test_date_with_self_paced_with_enrollment_before_course_start(self):
         course start date. """
         global_config = DynamicUpgradeDeadlineConfiguration.objects.create(enabled=True)
         course = create_self_paced_course_run(days_till_start=3)
+        course_overview = CourseOverview.load_from_module_store(course.id)
         overview = CourseOverview.get_from_id(course.id)
         expected = overview.start + timedelta(days=global_config.deadline_days)
-        enrollment = CourseEnrollmentFactory(course_id=course.id, mode=CourseMode.AUDIT)
+        enrollment = CourseEnrollmentFactory(course_id=course.id, mode=CourseMode.AUDIT, course=course_overview)
```

Comment:
I dislike that we have to override both course_id and course in order to get the `CourseEnrollmentFactory` to behave properly. Otherwise it creates a dummy course overview that is not a mirror of the course that was created earlier in the test.

I couldn't figure out a cleaner way to handle this, but very open to ideas!